messaging/sqlite_persistence.go

```diff
@@ -0,0 +1,64 @@
+package messaging
+
+import (
+	"database/sql"
+
+	"github.com/pkg/errors"
+	mvdsmigrations "github.com/status-im/mvds/persistenceutil"
+
+	"github.com/status-im/status-go/messaging/internal"
+	"github.com/status-im/status-go/messaging/layers/encryption"
+	encryptionmigrations "github.com/status-im/status-go/messaging/layers/encryption/migrations"
+	"github.com/status-im/status-go/messaging/types"
+	"github.com/status-im/status-go/sqlite"
+)
+
+/*
+ Reference implementation of the Persistence interface using SQLite.
+
+ Currently, only EncryptionPersistence is supported.
+ Other components remain in the Status client and may be migrated here as needed.
+*/
+
+// SQLiteMigrate applies necessary migrations to the SQLite database schema.
+func SQLiteMigrate(database *sql.DB, version uint) error {
+	if version > 0 {
+		err := createMigrationTables(database, version)
```

Comment:
The logic with `version` seem to be quite convoluted here and in `LatestMigrationUpToVersion` 🤔 

But I see `GetLastMigrationVersion` returns `migrationTableExists` here, which we ignore:
https://github.com/status-im/status-go/blob/78d6e84171656faf9c4001cb8151501f622321bc/protocol/sqlite/db.go#L19

Maybe just check it and create the tables if they don't exist?